### ISSUE TITLE
C# | New Special Camera Node for writing Multiplayer with separate dedicated server

### ISSUE THREAD

**Issue by adainx** [godotengine/godot-proposals#6375]
### Describe the project you are working on

I was trying to make a proof-of-concept project for myself. It is simple game where I wrote C# Dedicated server **using System.Net** and **System.Net.Sockets** (It's not Godot's ENEt system!!). I'm using Godot 4.0 beta 17 mono build, Client is C# too. Everything works perfectly, each players can control themselver expect one problem. I tried many solutions to overcome this problem (for example: making current for player camera, but from Godot's DOCs we know that: Only one camera can be set as the current).

### Describe the problem or limitation you are having in your project

When new player connects, the other players that already on server start to look through the camera of new one connected. New camera become current (Camera3D.MakeCurrent();). Players still control their Player.tscn, but only one camera available to be set. This problem doesn't exist in Godot's Enet Multiplayer, but I like I said before I didn't want to use Godot's Enet MP System.

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

I wonder whether it's possible to create new camera node that will allow create multiplayer where player can have each camera for themself.

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

When creating and spawning new player, he will use new camera node that wont change to new one. So every player will use their own camera in their Player.TSCN

### If this enhancement will not be used often, can it be worked around with a few lines of script?

Godot starts to become more and more popular, and I think a lot of developer will start to look to create their multiplayers projects using Godot.

### Is there a reason why this should be core and not an add-on in the asset library?

This is about adding new Camera node. If it's possible to made as an add-on it's cool. But I didn't find anything, I think it's something that should be and work out-of-box.

Thanks!

**Comment by AThousandShips** [comment]
Are you running both players on the same window? Are you making a split screen project?

**Comment by adainx** [comment]
> Are you running both players on the same window? Are you making a split screen project?

No, I run it in multiple instances. But I also tried exporting project and running it from created game .EXE. No, it's not split screen, I tried to make simple multiplayer project. If I misunderstood Your questions, please tell me about it.

**Comment by AThousandShips** [comment]
You probably need to use replication and spawning, as you can absolutely have different cameras in networking, but hard to tell without knowing how you actually make the networking work

**Comment by adainx** [comment]
> You probably need to use replication and spawning, as you can absolutely have different cameras in networking, but hard to tell without knowing how you actually make the networking work

When player spawns, in the root node of 3d scene and in `remote` tab in Godot appears new Player scenes which is (PackedScene), and when new node appears previous player node lose it's `current` check. I can try to make more screenshots or set Server and Client repository visibility to public, if it helps. But the problem is definitely with the camera, it's written in Godot's Docs that only one camera can be `current`. Indeed, while the game was running I tried to recheck `current` setting cameras from one player node to another and it return to be checked again and not checked to the first player. I also tried to make it from code in method where player spawn to set `_player.camera3d.MakerCurrent();`, but it only helped me to make cameras set to the last player connected. _Sorry my English might not be perfect_.
The reason why I decided to create this proposal is because I tried everything that is possible and nothing helped me. Most of the solution is for Godot ENEt Multiplayer System.

**Comment by AThousandShips** [comment]
Why are you setting all cameras to current? You should only make the camera of the player current, not of the remote pawns, on each client only one of the nodes is the player?

**Comment by adainx** [comment]
I do not set all cameras to current (but funny to notice I tried it too). I was trying to make current for new client and for his player scene when spawns in the world. Yes, on each client only one of the nodes is the player. It works only for one player (client), not for all, if it didn't I think players wont be able to control their `CharacterBody3D`, but everything is good with controlling (movement).

**Comment by AThousandShips** [comment]
That is very strange, setting the correct camera as current in each client should work, but I can't tell much without actually seeing the method you're using

**Comment by adainx** [comment]
Like I said before I can set either Server or Client repository to public. It's very simple projects. I'm just thinking, that You might don't want to spend time looking for the code and how it works OR I can just put screenshot of Player.TSCN-scene nodes structure and put the code contain player spawning, later (later because it's night to me now).
